Lib/test/test_urlparse.py

```diff
@@ -890,13 +890,34 @@ def test_parse_qsl_max_num_fields(self):
             urllib.parse.parse_qs(';'.join(['a=a']*11), max_num_fields=10)
         urllib.parse.parse_qs('&'.join(['a=a']*10), max_num_fields=10)
 
+    def test_urlencode(self):
+        result = urllib.parse.urlencode({'a': 1, 'b': None})
+        # bpo-18857: urlencode of a None value uses the string 'None'
+        assert set(result.split('&')) == {'a=1', 'b=None'}
+
+    def test_urlencode_standalone_keys(self):
+        result = urllib.parse.urlencode(
+            query={'a': 1, 'b': None},
+            standalone_keys=True
+        )
+        assert set(result.split('&')) == {'a=1', 'b'}
+
     def test_urlencode_sequences(self):
         # Other tests incidentally urlencode things; test non-covered cases:
         # Sequence and object values.
-        result = urllib.parse.urlencode({'a': [1, 2], 'b': (3, 4, 5)}, True)
+        result = urllib.parse.urlencode({'a': [1, 2], 'b': (3, 4, 5), 'c': None}, True)
         # we cannot rely on ordering here
```

Comment:
This comment is not correct anymore, now that dicts keep their insertion order we should be able to rely on ordering

Reply:
While this does seem true at the moment, the notes in the [Python 3.6 changelog](https://docs.python.org/3/whatsnew/3.6.html#new-dict-implementation) mention that the insertion order should currently be considered an implementation detail and is not (yet) safe to rely on.  It may be safer to keep the existing comment until there's a stronger guarantee around preserved insertion order.

Reply:
Ah, I [stand corrected](https://mail.python.org/pipermail/python-dev/2017-December/151283.html), sorry!  It looks like this behaviour was standardized in Python 3.7.  Given that, your suggestion sounds good - I'll take a moment to pause and read further before making applying this change.

Reply:
Yes, in Python 3.6 the dict started to keep their insertion order but it was not official. It was made a part of the language in Python 3.7 and in Python 3.8 they were made reversable.

Reply:
Change applied; thank you.